crmprtd/insert.py

```diff
@@ -54,6 +54,22 @@ def max_power_of_two(num):
     return 2 ** floor(mathlog(num, 2))
 
 
+def get_network_name(sesh, obs):
```

Comment:
This function is used *very* often. It's used in two tight loops that would iterate over all of the observations. Since it reaches out to the database, this is going to involve a lot of round trips (where are several orders of magnitude slower than anything happening in RAM).

I think that you'll *really* want to take a look at the `cached_function` decorator that's defined in `align.py`.